pkg/controller/machineautoscaler/machinetarget_test.go

```diff
@@ -375,3 +377,100 @@ func TestWarningForInvalidGPUAcceleratorLabel(t *testing.T) {
 		})
 	}
 }
+
+// Testing updating multiple annotations and adding new upstream annotations when the old ones exist
+func TestUpdatingScaleFromZeroAnnotations(t *testing.T) {
+	testConfigs := []struct {
+		name                  string
+		expectedNewAnnotation map[string]string
+		suppliedAnnotations   map[string]string
+	}{
+		{
+			name: "Supplying new CPU and Memory annotations",
+			expectedNewAnnotation: map[string]string{
+				annotationsutil.CpuKey:    "1",
+				annotationsutil.MemoryKey: "4Gi",
+			},
+			suppliedAnnotations: map[string]string{
+				annotationsutil.CpuKey:    "1",
+				annotationsutil.MemoryKey: "4Gi",
+			},
+		},
+		{
+			name: "Supplying old CPU annotation and adding upstream annotations",
+			expectedNewAnnotation: map[string]string{
+				annotationsutil.CpuKey:           "1",
+				annotationsutil.CpuKeyDeprecated: "1",
+			},
+			suppliedAnnotations: map[string]string{
+				annotationsutil.CpuKeyDeprecated: "1",
+			},
+		},
+		{
+			name: "Supplying old GPU annotation and adding upstream annotations",
+			expectedNewAnnotation: map[string]string{
+				annotationsutil.GpuCountKey:           "1",
+				annotationsutil.GpuCountKeyDeprecated: "1",
+				annotationsutil.GpuTypeKey:            "1",
```

Comment:
just to make this a little more realistic, i would change this value to `nvidia.com/gpu`, that is how it will look in a live cluster.